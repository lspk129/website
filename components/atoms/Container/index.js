import styled from 'styled-components';

const Container = styled.div`
  ${({ display }) => display && `display: ${display}`}
  ${({ textAlign }) => textAlign && `text-align: ${textAlign}`};
  ${({ alignSelf }) => alignSelf && `align-self: ${alignSelf}`};
  ${({ alignItems }) => alignItems && `align-items: ${alignItems}`};

  /* margin properties */
  ${({ m }) => m && `margin: ${m}rem;`}
  ${({ mx }) => mx && `margin: 0rem ${mx}rem;`}
  ${({ my }) => my && `margin: ${my}rem 0rem;`}
  ${({ mt }) => mt && `margin-top: ${mt}rem;`}
  ${({ mb }) => mb && `margin-bottom: ${mb}rem;`}
  ${({ mr }) => mr && `margin-right: ${mr}rem;`}
  ${({ ml }) => ml && `margin-left: ${ml}rem;`}

  /* padding properties */
  ${({ p }) => p && `padding: ${p}rem;`}
  ${({ px }) => px && `padding: 0rem ${px}rem;`}
  ${({ py }) => py && `padding: ${py}rem 0rem;`}
  ${({ pt }) => pt && `padding-top: ${pt}rem;`}
  ${({ pb }) => pb && `padding-bottom: ${pb}rem;`}
  ${({ pr }) => pr && `padding-right: ${pr}rem;`}
  ${({ pl }) => pl && `padding-left: ${pl}rem;`}

  /* background properties */
  ${({ bg, src }) => bg && `
    background-image: url(${src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
  ${({ backgroundColor }) => backgroundColor && `
    background-color: ${backgroundColor};
  `}
  ${({ zIndex }) => zIndex && `z-index: ${zIndex};`}
`;

export default Container;
