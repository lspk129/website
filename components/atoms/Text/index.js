import { createElement } from 'react';
import styled from 'styled-components';

const Text = styled(({ tag, hero, upperCase, children, ...props }) => createElement(tag, props, children))`
  text-align: center;

  ${({ tag, theme }) => tag === 'a' && `
    text-decoration: none;
    cursor: pointer;
    color: inherit;
    &:hover { color: ${theme.main}}
  `}

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ upperCase }) => upperCase && 'text-transform: uppercase'};
`;

Text.defaultProps = {
  tag: 'span',
};

export default Text;
