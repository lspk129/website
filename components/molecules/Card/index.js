import React from 'react';
import PropTypes from 'prop-types';

import Division from '../../atoms/Division';
import Title from '../../atoms/Title';

const Card = ({ image, label, text }) => (
  <Division textAlign={'center'} py={3} pr={1} pl={1}>
    {image}
    <Title tag={'h3'}>{label}</Title>
    <Title tag={'p'}>{text}</Title>
  </Division>
);

Card.propTypes = {
  image: PropTypes.node,
  label: PropTypes.string,
  text: PropTypes.string,
};

Card.defaultProps = {
  image: '',
  label: '',
  text: '',
};

export default Card;
