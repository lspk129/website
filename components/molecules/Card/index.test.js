import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card from './';
import Container from '../../atoms/Container';
import Text from '../../atoms/Text';

describe('Card', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toBePresent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with image props', () => {
    const wrapper = mount(<Card image={<img src={'../../../static/React_logo.png'} alt={'React'} />} />);
    expect(wrapper).toHaveProp('image');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with label props', () => {
    const wrapper = mount(<Card label={'test label'} />);
    expect(wrapper).toHaveProp('label', 'test label');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with text props', () => {
    const wrapper = mount(<Card text={'testing text'} />);
    expect(wrapper).toHaveProp('text', 'testing text');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Container component', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper.find(Container)).toBePresent();
  });

  it('renders Text component', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper.find(Text)).toBePresent();
  });
});
