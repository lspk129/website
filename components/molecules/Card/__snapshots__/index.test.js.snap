// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<styled.div
  pl={1}
  pr={1}
  py={3}
  textAlign="center"
>
  <Styled(Component)
    tag="h3"
  />
  <Styled(Component)
    tag="p"
  />
</styled.div>
`;

exports[`Card renders with image props 1`] = `
<Card
  image={
    <img
      alt="React"
      src="../../../static/React_logo.png"
    />
  }
  label=""
  text=""
>
  <styled.div
    pl={1}
    pr={1}
    py={3}
    textAlign="center"
  >
    <div
      className="sc-bdVaJa jJDreq"
    >
      <img
        alt="React"
        src="../../../static/React_logo.png"
      />
      <Styled(Component)
        tag="h3"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="h3"
        >
          <h3
            className="sc-bwzfXH elPfnD"
          />
        </Component>
      </Styled(Component)>
      <Styled(Component)
        tag="p"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="p"
        >
          <p
            className="sc-bwzfXH elPfnD"
          />
        </Component>
      </Styled(Component)>
    </div>
  </styled.div>
</Card>
`;

exports[`Card renders with label props 1`] = `
<Card
  image=""
  label="test label"
  text=""
>
  <styled.div
    pl={1}
    pr={1}
    py={3}
    textAlign="center"
  >
    <div
      className="sc-bdVaJa jJDreq"
    >
      <Styled(Component)
        tag="h3"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="h3"
        >
          <h3
            className="sc-bwzfXH elPfnD"
          >
            test label
          </h3>
        </Component>
      </Styled(Component)>
      <Styled(Component)
        tag="p"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="p"
        >
          <p
            className="sc-bwzfXH elPfnD"
          />
        </Component>
      </Styled(Component)>
    </div>
  </styled.div>
</Card>
`;

exports[`Card renders with text props 1`] = `
<Card
  image=""
  label=""
  text="testing text"
>
  <styled.div
    pl={1}
    pr={1}
    py={3}
    textAlign="center"
  >
    <div
      className="sc-bdVaJa jJDreq"
    >
      <Styled(Component)
        tag="h3"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="h3"
        >
          <h3
            className="sc-bwzfXH elPfnD"
          />
        </Component>
      </Styled(Component)>
      <Styled(Component)
        tag="p"
      >
        <Component
          className="sc-bwzfXH elPfnD"
          tag="p"
        >
          <p
            className="sc-bwzfXH elPfnD"
          >
            testing text
          </p>
        </Component>
      </Styled(Component)>
    </div>
  </styled.div>
</Card>
`;
